File: 02-java-exercises/5-anagram-java/src/Anagram.java

import java.util.Arrays;

public class Anagram {

    public static void main(String[] args) {

        String wordOne = "Rita";
        String wordTwo = "aitr";

        System.out.println(wordOne + " and " + wordTwo + " are anagrams: " + isAnagram(wordOne, wordTwo));
    }

    /* ANAGRAM */

    public static boolean isAnagram(String strOne, String strTwo) {
      strOne = strOne.toLowerCase();
      strTwo = strTwo.toLowerCase();
      
      strOne = sortString(strOne);
      strTwo = sortString(strTwo);
        
      if (strOne.length() != strTwo.length()) {
          return false;
      }
      if (!strOne.equals(strTwo)) {
        return false;
      }
      return true;
    }

    private static String sortString(String strToSort) {
      char[] charArray = strToSort.toCharArray();
      Arrays.sort(charArray);
      return new String (charArray);
    }
}

File: 01-js-exercises/5-anagram-js/anagram.js
/* ANAGRAM */
console.log("// Anagram //");

var word1 = "ana";
var word2 = "naa"

function isAnagram(word1, word2){
  
  var word1Sorted = word1.toLowerCase().split("").sort().join("");
  var word2Sorted = word2.toLowerCase().split("").sort().join("");
  
  if (word1Sorted !== word2Sorted) {
    return false;
  }
  return true;
 
}

console.log(word1 + " and "+ word2 + " are anagrams: " + isAnagram(word1, word2) +".");

